File: exercism/typescript/word-count/word-count.ts
class Words {
    count(sentence: string) {
        const words = sentence.trim().split(/\s+/)
        const counter = new Map<string, number>()
        words.map(elt=>elt.toLowerCase()).forEach(elt => counter.set(elt, counter.has(elt) ? <number>counter.get(elt) + 1: 1))
        return counter
    }
  }
  
  export default Words

File: exercism/elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    String.split(String.downcase(sentence),~r/[^[:alnum:]-]+/u, trim: true)
    |> Enum.frequencies()
  end
end
